sched/queue/memory/simple.go

```diff
@@ -48,33 +142,28 @@ func (q *simpleQueue) Enqueue(job sched.Job) (string, error) {
 		return "", err
 	}
 	resultCh := make(chan enqueueResult)
-	q.inCh <- enqueueReq{job, resultCh}
+	q.reqCh <- enqueueReq{job, resultCh}
 	result := <-resultCh
 	return result.jobID, result.err
 }
 
-func (q *simpleQueue) Close() error {
-	close(q.inCh)
-	close(q.outCh)
-	return nil
+type statusResult struct {
+	status sched.JobStatus
+	err    error
 }
 
-func (q *simpleQueue) loop() {
-	for {
-		req, ok := <-q.inCh
-		if !ok {
-			// Channel is cloed
-			return
-		}
-		job := req.job
-		id := fmt.Sprintf("%v", q.nextID)
-		q.nextID++
-		job.Id = id
-		select {
-		case q.outCh <- simpleWorkItem(job):
-			req.resultCh <- enqueueResult{id, nil}
-		default:
-			req.resultCh <- enqueueResult{"", queue.NewCanNotScheduleNow(1*time.Second, "queue full")}
-		}
-	}
+func (q *simpleQueue) Status(jobId string) (sched.JobStatus, error) {
```

Comment:
Just a thought, we can pull out the send request logic. This is uglier but pulls out doRequest() rather than copy-pasting core logic for each additional request type.
```
type request interface {
  resultCh() *(chan enqueueResult)
}

type req struct {
	resultCh chan enqueueResult
}
func (req *request) resultCh() *(chan enqueueResult) {
   return &resultCh
}

type enqueueReq struct {
        req
	job      sched.Job
}
type statusReq struct {
        req
	jobId    string
}

func (q * simpleQueue) doRequest(req request) interface{} {
	*req.resultCh := make(chan interface{})
	q.reqCh <- req
	return <-req.resultCh
}

func (q *simpleQueue) Enqueue(job sched.Job) (string, error) {
	err := queue.ValidateJob(job)
	if err != nil {
		return "", err
	}
        result := q.doRequest(enqueueReq{job, resultCh}).(*enqueueResult)
        return result.jobID, result.err
}
```